lib/internal/fs/promises.js

```diff
@@ -273,8 +274,17 @@ function checkAborted(signal) {
     throw new AbortError();
 }
 
-async function writeFileHandle(filehandle, data, signal) {
-  // `data` could be any kind of typed array.
+async function writeFileHandle(filehandle, data, signal, encoding) {
+  checkAborted(signal);
+  if (isCustomIterable(data)) {
+    for await (const buf of data) {
+      checkAborted(signal);
+      await write(
+        filehandle, buf, undefined, isArrayBuffer(buf) ? buf.length : encoding);
```

Comment:
```suggestion
        filehandle, buf, undefined, isArrayBufferView(buf) ? buf.length : encoding);
```
Can you please add tests for:
- when the the `Iterable` yields `Buffer`s.
- when the the `Iterable` yields invalid objects (`null`, `undefined`, ...).
- non-UTF8 string encoding.

Sorry I missed that in the earlier reviews, let me know if you want help writing those tests.

Reply:
@aduh95 
Thanks for the review.
Some tests have been added.

> non-UTF8 string encoding.

Sorry, I don't know how to write the above test.
Is there any code within nodejs that I can refer to?

Reply:
You could make something similar to
https://github.com/nodejs/node/blob/44243e59df293c3579eb1dc779d5c2aaefb4d8bd/test/parallel/test-fs-write.js#L44-L82

Reply:
Thanks!
I've added a test for reference.